File: src/main/scala/com/kotori316/fluidtank/network/ServerProxy.java
package com.kotori316.fluidtank.network;

import java.util.Optional;

import net.minecraft.world.entity.Entity;
import net.minecraft.world.item.Item;
import net.minecraft.world.level.Level;
import net.minecraftforge.network.NetworkEvent;

public class ServerProxy extends SideProxy {

    @Override
    public Optional<Level> getLevel(NetworkEvent.Context context) {
        return Optional.ofNullable(context.getSender()).map(Entity::getCommandSenderWorld);
    }

    @Override
    public Item.Properties getTankProperties() {
        return new Item.Properties();
    }

    @Override
    public Item.Properties getReservoirProperties() {
        return new Item.Properties();
    }
}

File: src/main/scala/com/kotori316/fluidtank/network/ServerProxy.scala
package com.kotori316.fluidtank.network

import com.kotori316.fluidtank.ModObjects
import net.minecraft.world.item.Item
import net.minecraft.world.level.Level
import net.minecraftforge.network.NetworkEvent

class ServerProxy extends SideProxy {
  override def getLevel(context: NetworkEvent.Context): Option[Level] = Option(context.getSender).map(_.getCommandSenderWorld)

  override def getTankProperties: Item.Properties = new Item.Properties().tab(ModObjects.CREATIVE_TABS)

  override def getReservoirProperties: Item.Properties = new Item.Properties().tab(ModObjects.CREATIVE_TABS)
}
